src/window_battlestatus.cpp

```diff
@@ -88,7 +88,7 @@ void Window_BattleStatus::Refresh() {
 			} else {
 				if (lcf::Data::battlecommands.battle_type == lcf::rpg::BattleCommands::BattleType_traditional) {
 					DrawActorState(*actor, 84, y);
-					contents->TextDraw(136 + 4 * 6, y, Font::ColorDefault, std::to_string(actor->GetHp()), Text::AlignRight);
+					contents->TextDraw(136 + 4 * 6, y, actor->GetHp() == 0 ? Font::ColorKnockout : actor->GetHp() <= actor->GetMaxHp() / 4 ? Font::ColorCritical : Font::ColorDefault, std::to_string(actor->GetHp()), Text::AlignRight);
```

Comment:
I agree with c1 that this ternary is hard to read.
https://github.com/EasyRPG/Player/blob/master/src/window_base.cpp#L177-L181

Reply:
We have this in 3 places now at least. Maybe is time to make this a real function.

Reply:
yeah.
Maybe add two new functions to window_base.h which simply draw the currect HP/MP value at cx/cy. There is already DrawActorHp but this one is not generic enough for this.
```
void DrawActorHpValue(const Game_Battler& actor, int cx, int cy) const;
void DrawActorSpValue(const Game_Battler& actor, int cx, int cy) const;
```

Reply:
Added the functions and replaced the ternary.

Reply:
The color logic is still duplicated though.

Reply:
yes looks good